input/system/azure/system.go

```diff
@@ -0,0 +1,282 @@
+			system.CPUStats["all"] = state.CPUStatistic{
+				DiffedOnInput: true,
+				DiffedValues: &state.DiffedSystemCPUStats{
+					UserPercent: *metricValue.Average,
+				},
+			}
+		case "memory_percent":
+			// Currently, we can only retrieve memory percent
+			// Since total memory size is not listed in the server info,
+			// we are unable to pass this value to pganalyze at the moment
+		case "txlogs_storage_used":
+			system.XlogUsedBytes = uint64(*metricValue.Average)
+		case "network_bytes_egress":
+			// value is total of 1min
+			diffedNetworkStats.TransmitThroughputBytesPerSecond = uint64(*metricValue.Average / 60)
+		case "network_bytes_ingress":
+			// value is total of 1min
+			diffedNetworkStats.ReceiveThroughputBytesPerSecond = uint64(*metricValue.Average / 60)
+		case "read_iops":
+			diffedDiskStats.ReadOperationsPerSecond = *metricValue.Average
+		case "write_iops":
+			diffedDiskStats.WriteOperationsPerSecond = *metricValue.Average
+		case "disk_queue_depth":
+			diffedDiskStats.AvgQueueSize = int32(*metricValue.Average)
+		case "read_throughput":
+			diffedDiskStats.BytesReadPerSecond = *metricValue.Average
+		case "write_throughput":
+			diffedDiskStats.BytesWrittenPerSecond = *metricValue.Average
+		case "storage_used":
+			if system.Info.Azure.StorageGB != 0 {
+				// Flexible Server
+				totalGB := uint64(system.Info.Azure.StorageGB)
+				system.DiskPartitions = make(state.DiskPartitionMap)
+				system.DiskPartitions["/"] = state.DiskPartition{
+					DiskName:   "default",
+					UsedBytes:  uint64(*metricValue.Average),
+					TotalBytes: totalGB * 1024 * 1024 * 1024,
+				}
+			} else if system.Info.Azure.CoordinatorStorageMB != 0 {
+				// Cosmos DB
+				totalMB := uint64(system.Info.Azure.CoordinatorStorageMB)
+				system.DiskPartitions = make(state.DiskPartitionMap)
+				system.DiskPartitions["/"] = state.DiskPartition{
+					DiskName:   "default",
+					UsedBytes:  uint64(*metricValue.Average),
+					TotalBytes: totalMB * 1024 * 1024,
+				}
+			}
+		}
+	}
+	system.NetworkStats = make(state.NetworkStatsMap)
+	system.NetworkStats["default"] = state.NetworkStats{
+		DiffedOnInput: true,
+		DiffedValues:  diffedNetworkStats,
+	}
+	system.Disks = make(state.DiskMap)
+	system.Disks["default"] = state.Disk{}
+	if strings.ToLower(system.Info.Azure.ResourceType) == "flexibleservers" {
+		// DiskStats is only available with Flexible Server
+		system.DiskStats = make(state.DiskStatsMap)
+		system.DiskStats["default"] = state.DiskStats{
+			DiffedOnInput: true,
+			DiffedValues:  diffedDiskStats,
+		}
+	}
+
+	server.SelfTest.MarkCollectionAspectOk(state.CollectionAspectSystemStats)
+
+	return
+}
+
+// getFirstMetricValue gets the first data from the time series metric and returns the value
+func getFirstMetricValue(metric *azquery.Metric) (metricValue *azquery.MetricValue) {
+	for _, timeSeriesElement := range metric.TimeSeries {
+		for _, mValue := range timeSeriesElement.Data {
+			metricValue = mValue
+		}
+	}
+	return
+}
```

Comment:
Reviewing #645, this definition jumped out at me as a little unclear: this seems to look for the last Data element in the last TimeSeries, right? That doesn't seem to align with the comment.

Reply:
Very good point, if I wanna make the comment right, I should be adding return after line 278 - otherwise, I should be updating the comment to say "the last data". I think the comment and function name is pretty clear that Keiko in September wanted to pick the first value, so I'll update the logic to do so.